sqlx-core/src/postgres/types/network.rs

```diff
@@ -0,0 +1,119 @@
+use std::convert::TryInto;
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use ipnetwork::{IpNetwork, Ipv4Network, Ipv6Network};
+
+use crate::decode::Decode;
+use crate::encode::Encode;
+use crate::postgres::protocol::TypeId;
+use crate::postgres::row::PgValue;
+use crate::postgres::types::PgTypeInfo;
+use crate::postgres::Postgres;
+use crate::types::Type;
+use crate::Error;
+
+#[cfg(windows)]
+const AF_INET: u8 = 2;
+// Maybe not used, but defining to follow Rust's libstd/net/sys
+#[cfg(redox)]
+const AF_INET: u8 = 1;
+#[cfg(not(any(windows, redox)))]
+const AF_INET: u8 = libc::AF_INET as u8;
+
+const PGSQL_AF_INET: u8 = AF_INET;
+const PGSQL_AF_INET6: u8 = AF_INET + 1;
+
+const INET_TYPE: u8 = 0;
+const CIDR_TYPE: u8 = 1;
+
+impl Type<Postgres> for IpNetwork {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::new(TypeId::INET, "INET")
+    }
+}
+
+impl Type<Postgres> for [IpNetwork] {
+    fn type_info() -> PgTypeInfo {
+        PgTypeInfo::new(TypeId::ARRAY_INET, "INET[]")
+    }
+}
+
+impl Encode<Postgres> for IpNetwork {
+    fn encode(&self, buf: &mut Vec<u8>) {
+        encode(self, INET_TYPE, buf)
+    }
+
+    fn size_hint(&self) -> usize {
+        match self {
+            IpNetwork::V4(_) => 8,
+            IpNetwork::V6(_) => 20,
+        }
+    }
+}
+
+impl<'de> Decode<'de, Postgres> for IpNetwork {
+    fn decode(value: Option<PgValue<'de>>) -> crate::Result<Self> {
+        match value.try_into()? {
+            PgValue::Binary(buf) => decode(buf, INET_TYPE),
+            PgValue::Text(s) => decode(s.as_bytes(), INET_TYPE),
```

Comment:
The TEXT mode format is ( and I'm guessing here ) probably: `192.168.100.128/25`. It's generally whatever `psql` shows when you `SELECT` a value of it.